### ISSUE TITLE
🌟Feature request: add more options to `closeOnOutsideClick` prop

### ISSUE THREAD

**Issue by anatolzak** [melt-ui/melt-ui#1150]
When dealing with multiple nested floating elements and passing `closeOnOutsideClick: false` on a child element, clicking outside is disabled.

It would be nice to be able to have an element not be counted as a floating element that should react to outside clicks. For example, the following is currently not possible: Having an open popover inside a dialog and on outside click, the dialog should close. Currently we can either pass `closeOnOutsideClick: true` to the popover, which will only close the popover on outside click. And if you pass `closeOnOutsideClick: false` to the popover, on outside click nothing will close.

**Describe the solution you'd like**
I suggest that we allow passing `closeOnOutsideClick: null` to not count an element in the interact-outside stack and let the parent element react to the outside interaction.

To recap, the new options for `closeOnOutsideClick` should be:
- `true`: on outside interaction, the element should close as long as it's the top-most floating element.
- `false`: on outside interaction, the element should not close.
- `null`: on outside interaction, the parent element should handle the interaction.

In https://github.com/melt-ui/melt-ui/pull/1080 we use a stack to maintain the top-most floating elements interested in outside interactions. the PR handles both `closeOnOutsideClick: true | false`. The only change that needs to be made is to simply not add an element to the stack provided `closeOnOutsideClick: null`.

**Comment by anatolzak** [comment]
See https://github.com/melt-ui/melt-ui/pull/1152 for videos showing how each option behaves
